GTG/gtk/ui/general_preferences.ui

```diff
@@ -0,0 +1,390 @@
+                </child>
+              </object>
+            </child>
+          </object>
+        </child>
+        <child>
+          <object class="GtkBox" id="bg_color_box">
+            <property name="margin">20</property>
+            <property name="margin-start">40</property>
+            <child>
+              <object class="GtkSwitch" id="bg_color_button">
+                <property name="visible">True</property>
+                <property name="valign">center</property>
+                <signal handler="on_bg_color_toggled" name="state-set" swapped="no"/>
+              </object>
+            </child>
+            <child>
+              <object class="GtkBox">
+                <property name="orientation">vertical</property>
+                <property name="valign">center</property>
+                <property name="vexpand">false</property>
+                <property name="visible">true</property>
+                <property name="margin-start">10</property>
+                <child>
+                  <object class="GtkLabel">
+                    <property name="label" translatable="yes">Enable coloured backgrounds</property>
+                    <property name="visible">true</property>
+                    <property name="xalign">0.0</property>
+                    <attributes>
+                      <attribute name="weight" value="bold"/>
+                    </attributes>
+                  </object>
+                </child>
+                <child>
+                  <object class="GtkLabel">
+                    <property name="label" translatable="yes">Show coloured backgrounds for a task based on the colours of its tags</property>
+                    <property name="visible">true</property>
+                    <property name="xalign">0.0</property>
+                    <style>
+                      <class name="dim-label"/>
+                    </style>
+                    <attributes>
+                      <attribute name="scale" value="0.833333"/>
+                    </attributes>
+                  </object>
+                </child>
+              </object>
+            </child>
+          </object>
+        </child>
+        <child>
+          <object class="GtkBox" id="time_box">
+            <property name="visible">True</property>
+            <property name="can_focus">False</property>
+            <property name="margin-bottom">20</property>
+            <property name="margin-start">40</property>
+            <child>
+              <object class="GtkEntry" id="time_entry">
+                <property name="visible">True</property>
+                <property name="can_focus">True</property>
+                <property name="width_chars">8</property>
+                <property name="max-width-chars">8</property>
+                <property name="max_length">8</property>
+                <property name="xalign">0.5</property>
+                <signal name="changed" handler="on_valid_check" swapped="no"/>
```

Comment:
Can you make it more clear that this is only for time? e.g. `on_valid_time_check`, `on_leave_time_entry`